src/net/rpc.rs

```diff
@@ -589,97 +589,94 @@ impl ConversationHttp {
-        // let tip_index_hash =
-        //     StacksBlockHeader::make_index_block_hash(tip_consensus_hash, tip_block_hash);
-
-        // for block_height in min_block_height..=max_block_height {
-        //     match StacksChainState::get_index_tip_ancestor(
-        //         &mut headers_tx,
-        //         &tip_index_hash,
-        //         block_height,
-        //     )? {
-        //         Some(header) => blocks_ids.push(header.index_block_hash()),
-        //         _ => {}
-        //     }
-        // }
-
-        // match atlasdb.get_attachments_available_at_pages_indexes(&pages_indexes, &blocks_ids) {
-        //     Ok(pages) => {
-        //         let pages = pages
-        //             .into_iter()
-        //             .zip(pages_indexes)
-        //             .map(|(inventory, index)| AttachmentPage { index, inventory })
-        //             .collect();
-
-        //         let content = GetAttachmentsInvResponse {
-        //             block_id: tip.clone(),
-        //             pages,
-        //         };
-        //         let response = HttpResponseType::GetAttachmentsInv(response_metadata, content);
-        //         response.send(http, fd)
-        //     }
-        //     Err(e) => {
-        //         let msg = format!("Unable to read Atlas DB");
-        //         warn!("{}", msg);
-        //         let response = HttpResponseType::ServerError(response_metadata, msg);
-        //         response.send(http, fd)?;
-        //         return Err(net_error::DBError(e));
-        //     }
-        // }
+        let tip = StacksBlockHeader::make_index_block_hash(&tip_consensus_hash, &tip_block_hash);
+        let mut pages_indexes = pages_indexes.iter().map(|i| *i).collect::<Vec<u32>>();
+        pages_indexes.sort();
+
+        let mut pages = vec![];
+
+        // We need to rebuild an ancestry tree, but we're still missing some informations at this point
+        for page_index in pages_indexes.iter() {
+            let (min_block_height, max_block_height) =
+                match atlasdb.get_minmax_heights_window_for_page_index(*page_index) {
+                    Ok(window) => window,
+                    Err(e) => {
+                        let msg = format!(
+                            "Unable to read min/max heights for page index {} - {}",
+                            page_index, e
+                        );
+                        warn!("{}", msg);
+                        let response = HttpResponseType::NotFound(response_metadata, msg);
+                        return response.send(http, fd);
+                    }
+                };
+
+            let mut blocks_ids = vec![];
+            let headers_conn = chainstate.index_conn()?;
+            let tip_index_hash =
+                StacksBlockHeader::make_index_block_hash(tip_consensus_hash, tip_block_hash);
+
+            for block_height in min_block_height..=max_block_height {
```

Comment:
This is going to be painfully slow, since you're doing a MARF read on each block height.  It might be faster (like, over 10x faster) if you simply loaded the header at `max_block_height-1`, and then iteratively loaded its ancestors' headers by (1) getting the parent Stacks block ID using `get_parent_block_id()`, and then using the parent block ID to load the parent via the `get_stacks_block_header_info_by_index_block_hash()` method.  This all is running in the network's critical path, so it's of paramount importance that this RPC method either be very very fast, or operate in a streaming manner (like how we serve blocks and microblocks) so it doesn't stall all the rest of the network I/O.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2453/commits/877471b1a277fbf0fbe5afcafaad94bd5b370434